### PR TITLE
#16 document platforms order states

### LINKED ISSUE #16: Document platforms' order states
Map order states used by platform and how they should implement them in the standard

Use gitbook to document it for now, but we need to come up with a better solution?

We can use this work to start improving the implementation documentation

### PR DESCRIPTION
**Pull request by RaggedStaff**
Closes #16 

Initial draft showing Platform Order State mappings. 

Not sure why it's including a bunch of changes that already appear in [master](../tree/master). :confused:

### CODE REVIEW

File: technical-specifications/order-states.md
Comment (RachL) [inline technical-specifications/order-states.md:1]: @RaggedStaff before the address state, the order is at the "cart" state in OFN. In fact, as soon as you land on a shop, an order is open for your user. I don't know if we want to follow this state as well?

regarding the DFC class, maybe it's just me but I found it weird to have order states separated from fulfillment / payment states (as fulfillment and payment state only happen on completed orders - do you see what I mean?)

Comment (RachL) [inline technical-specifications/order-states.md:1]: Regarding payment state, were did you found the OFN ones? 

In the db I see for OFN these payment states:
- paid
- failed
- credit_owed
- balance_due
- requires_authorization
- void

Comment (RachL) [inline technical-specifications/order-states.md:1]: For shipping states, I see in the db:

- backorder
- canceled
- partial
- pending
- ready
- shipped

**Review by RachL — CHANGES_REQUESTED** [review]

**Comment by sankarom** [comment]
Hi, For order states, this seems good because Socleo has : 

- draft
- pending (payment or other) = held ?
- completed
- retracted (by the customer)
- cancelled (by the seller)

no real payment status for us, either it is payed or it isn't

and i'm not sure what the Fulfilment States correspond to? what does "processed" order mean?
logistical states ? in this case we have :

- in receipt
- received
- in preparation
- prepared
- in delivery
- delivered

File: technical-specifications/order-states.md
Comment (RaggedStaff) [reply technical-specifications/order-states.md:1]: It was a long time ago, but I think from @mkllnk

Comment (mkllnk) [reply technical-specifications/order-states.md:1]: Yes, that document lists the OFN states that have been bundled into the DFC states. For example, `pending` and `processing` are important for OFN while processing the payment but that state doesn't have to be communicated to others. So while a payment is processed, the API will still tell you that the payment is `unpaid`.

**Review by RachL — CHANGES_REQUESTED** [review]
@RaggedStaff we are missing shipment/cancelled as OFN euivalent? Otherwise I think we are good :+1:

**Review by RachL — APPROVED** [review]

**Review by lecoqlibre — COMMENTED** [review]
OK for me but a little typo is in Order States > Complete. "Order is finalied" should be "Order is finalized" instead (finalized).

**Review by lecoqlibre — APPROVED** [review]

**Review by RachL — APPROVED** [review]

**Pull request merged**